WooCommerce/Classes/ViewRelated/Orders/Refund/RefundSubmissionUseCase.swift

```diff
@@ -286,11 +301,14 @@ private extension RefundSubmissionUseCase {
                     self.attemptCardPresentRefund(refundAmount: refundAmount, charge: charge, onCompletion: onCompletion)
                 case .failure(let cancelError):
                     // Informs that payment can't be retried.
-                    self.alerts?.nonRetryableError(from: self.rootViewController, error: cancelError)
+                    self.alerts.nonRetryableError(from: self.rootViewController, error: cancelError)
                     onCompletion(.failure(error))
                 }
             }
-        }
+        }, dismissError: { viewController in
+            viewController?.dismiss(animated: true)
+            onCompletion(.failure(error))
+        })
```

Comment:
There will be a merge conflict with #6654 with this, but it should be easily resolved

Reply:
I can help with the merge conflicts after this PR is merged!